src/docker.rs

```diff
@@ -204,30 +198,40 @@ WORKDIR /home/solana
         Ok(())
     }
 
-    pub fn push_image(docker_image: &DockerImage) -> Result<(), Box<dyn Error>> {
+    pub fn push_image(docker_image: &DockerImage) -> Result<(), Box<dyn Error + Send>> {
         let progress_bar = new_spinner_progress_bar();
         progress_bar.set_message(format!(
-            "{ROCKET}Pushing {} image to registry...",
-            docker_image.validator_type()
+            "{ROCKET}Pushing {docker_image} image to registry...",
         ));
-        let command = format!("docker push '{}'", docker_image);
-        let output = match Command::new("sh")
+        let command = format!("docker push '{docker_image}'");
+        let output = Command::new("sh")
             .arg("-c")
             .arg(&command)
             .stdout(Stdio::null())
             .stderr(Stdio::null())
             .spawn()
             .expect("Failed to execute command")
             .wait_with_output()
-        {
-            Ok(res) => Ok(res),
-            Err(err) => Err(Box::new(err) as Box<dyn Error>),
-        }?;
+            .expect("Failed to push image");
 
         if !output.status.success() {
-            return Err(output.status.to_string().into());
+            return Err(Box::new(DockerPushThreadError::from(
+                output.status.to_string(),
+            )));
         }
         progress_bar.finish_and_clear();
         Ok(())
     }
+
+    pub fn push_images<'a, I>(&self, validators: I) -> Result<(), Box<dyn Error + Send>>
+    where
+        I: IntoIterator<Item = &'a Validator>,
+    {
+        info!("Pushing images...");
+        validators
+            .into_iter()
+            .collect::<Vec<_>>() // Collect into Vec and thread push
+            .par_iter()
+            .try_for_each(|validator| Self::push_image(validator.image()))
```

Comment:
Since `push_image` is using `Command::spawn` internally, it creates a new process anyway, so the `par_iter` is a little overkill. You could have `push_image` return the `process::Child` and then iterate through them here with `wait_with_output`, and completely remove rayon